packages/tools/examples/contourApi/index.ts

```diff
@@ -0,0 +1,530 @@
+  )
+  .addInstruction(
+    'Decimate Polylines: Simplifies polylines by removing points using Ramer-Douglas-Peucker algorithm with epsilon tolerance of 2.0 units. Shows reduction percentage.'
+  );
+
+addButtonToToolbar({
+  title: 'Find Contour Holes',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
+
+    if (annotations.length === 0) {
+      console.log('No annotations found. Please draw some contours first.');
+      return;
+    }
+
+    // Collect all polylines from all annotations
+    const allPolylines = [];
+    const annotationPolylineMap = new Map();
+
+    annotations.forEach((annotation, index) => {
+      if (annotation?.data?.contour?.polyline) {
+        const polyline = getAnnotationPolyline(annotation);
+        allPolylines.push(polyline);
+        annotationPolylineMap.set(polyline, annotation);
+        console.log(`Annotation ${index + 1}:`);
+        console.log('Original polyline points:', polyline.length);
+      }
+    });
+
+    if (allPolylines.length === 0) {
+      console.log('No valid polylines found.');
+      return;
+    }
+
+    try {
+      // Find holes in all contours
+      const holesResult = findContourHoles(allPolylines);
+      console.log('Holes found:', holesResult);
+
+      // Check if any holes were found and change annotation color to red
+      if (holesResult && holesResult.length > 0) {
+        console.log('Holes detected! Changing annotation colors to red.');
+        holesResult.forEach((holeResult) => {
+          holeResult.holeIndexes.forEach((holeIndex) => {
+            const polyline = allPolylines[holeIndex];
+            const annotation = annotationPolylineMap.get(polyline);
+            if (annotation) {
+              // Set annotation style to red
+              const styles = {
+                color: 'rgb(255, 0, 0)', // Red color
+              };
+
+              csToolsAnnotation.config.style.setAnnotationStyles(
+                annotation.annotationUID,
+                styles
+              );
+            }
+          });
+          // select the polyline with holes
+          const polyline = allPolylines[holeResult.contourIndex];
+          const annotation = annotationPolylineMap.get(polyline);
+          if (annotation) {
+            csToolsAnnotation.selection.setAnnotationSelected(
+              annotation.annotationUID
+            );
+          }
+        });
+
+        // Re-render to show the color changes
+        const renderingEngine = getRenderingEngine(renderingEngineId);
+        renderingEngine.render();
+      }
+    } catch (error) {
+      console.error('Error finding contour holes:', error);
+    }
+  },
+});
+
+addButtonToToolbar({
+  title: 'Supersample Polylines',
+  onClick: () => {
+    const annotations = cornerstoneTools.annotation.state.getAllAnnotations();
```

Comment:
same 

`cornerstoneTools.segmentation.utilities.smoothContours({ segmentationId, segmentIndex}, options: {epsilon, mode = "linear" })`

we already have all the functionality for it 

smoothAnnotation in `packages/tools/src/utilities/planarFreehandROITool/smoothAnnotation.ts` and `getInterpolatedPoints` in `packages/tools/src/utilities/planarFreehandROITool/smoothPoints.ts` use them